### ISSUE TITLE
Complex rrules and frules

### ISSUE THREAD

**Issue by CheukHinHoJerry** [ACEsuit/Polynomials4ML.jl#41]
Currently the rrule of ScalarPoly4MLBasis is only implemented for real inputs. We should generalize it to Complex later.

**Comment by cortner** [comment]
I agree. Can you please remove the rrule tests for all complex models, including also the complex spherical and solid harmonics

**Comment by CheukHinHoJerry** [comment]
Sure. I will push an update soon after doing that and make sure the CI test passes. Most of the issue that you mentioned in the PR is already fixed.

**Comment by cortner** [comment]
save for CYlm and CRlm??

**Comment by CheukHinHoJerry** [comment]
> save for CYlm and CRlm??

I will look into it today but I am not sure when I can fix that. Shall we merge the branch first and do that with a small PR later?

**Comment by cortner** [comment]
No rush - let's merge first

**Comment by cortner** [comment]
I believe this is still open. At some point it would be good if somebody tests this and starts fixing it.

**Comment by cortner** [comment]
won't fix unless there is an urgent concern for a concrete project.
